src/client/datascience/jupyter/jupyterConnection.ts

```diff
@@ -243,8 +241,6 @@ class JupyterConnection implements IJupyterConnection {
             childProc.on('exit', (c) => {
                 // Our code expects the exit code to be of type `number` or `undefined`.
                 const code = typeof c === 'number' ? c : 0;
-                this.valid = false;
-                this.localProcExitCode = code;

```

Comment:
All of this was only required in `variable` viewer.
Var viewer should no longer need this, none of the other places (intellisense, kernel execution, etc rely on these).
Hence hope is to use a standard way to detect whether the process/kernel is dead